src/EnergyPlus/UtilityRoutines.cc

```diff
@@ -589,6 +589,13 @@ namespace UtilityRoutines {
 
 } // namespace UtilityRoutines
 
+int getEnumerationValue(gsl::span<std::string_view> sList, std::string_view s)
+{
+    for (unsigned int i = 0; i < sList.size(); ++i)
+        if (UtilityRoutines::SameString(sList[i], s)) return i;
+    return -1;
+}
+
```

Comment:
Is this the function that's supposed to get used to match input enum values against the accepted values?

Reply:
Yes. Create a `constexpr std::array<std::string_view>` that has the strings corresponding to the enumeration and pass it to this function. The use of `gsl::span` (the artist known as `std::span` in C++20) allows you to pass `std::array` of different sizes to the same function without templating. The same array can also be used to convert the enumeration back to string using direct lookup.

Reply:
OK. In the input case, shouldn't this `assert` or something similar rather than return -1? Since the input values are validated by the input processor, so if the value isn't found there must be a mismatch between what's in the input schema and what's in the array.

Reply:
So, it should return -1 (@jasondegraw, this is not the hill you want to die on). Whether it should also do something else is debatable. 

EnergyPlus generally doesn't `assert` silently, it calls `ShowSevereError` or `ShowFatalError`. For input processing, that error message usually has some context information like the object type and name and field name. To preserve this kind of diagnostic, we would need to pass this context information to the function. Which we could do. Or we could create a wrapper function that takes this additional context and also prints a message. Or we could keep that logic at the call sites.

Reply:
The tradeoff above is setting up and passing additional arguments for the rare case in which there is an error on one hand, and decluttering call-sites on the other. 

There's a larger discussion to be had about error handling in EnergyPlus. Right now, the paradigm is to "accumulate" and track errors using `ErrorFound` parameters that are threaded through many functions, and then at various points to check that parameter and call `ShowFatalError` and give up the ghost. At some point we may want to switch to using exceptions. "Accumulating" errors is a bit trickier in that case, but it could be made to work. We would have to be careful to not throw exceptions across the API.

Reply:
Not looking for a place to die. If the user has put in a value that doesn't match the available choices in the schema, they'll have already gotten an error:

```
** Severe  ** <root>[AirflowNetwork:SimulationControl][House AirflowNetwork][wind_pressure_coefficient_type] - "NoWindAtAll" - Failed to match against any enum values.
**  Fatal  ** Errors occurred on processing input file. Preceding condition(s) cause termination.
```

If processing has made it to the GetInput stage, I'm pretty sure it's a developer error and not a user error. Moving on.

Reply:
Is that done by the JSON validator?

Reply:
Yes

Reply:
OK, I did not know that. In part because there are still a bunch of places in the `GetInputData` routines that `ShowSevereError` for the `else` case of converting a string to an `enum`. If these `GetInputData` cases are never hit anymore, then yes we can `assert`. Thanks for the clarification.

At this point I am wondering if we can actually use the strings from the parsed IDD file to do this conversion, but it feels like this will be much more trouble than it is worth.

Reply:
@amirroth As a history lesson, prior to the new epJSON-based input processing, key choices were not validated until the `GetInput` routines. Until then, the key choices in the IDD only informed IDF Editor and other tools. So, nearly every `GetInput` function still has these error cases in place. And some newer `GetInput` functions probably have them, too, just because.

Reply:
I definitely remember the pre-epJSON days, it hasn't been around for that long. What I didn't realize was that this validation had moved to epJSON and that this error-handling code had been largely removed.

Reply:
> . . . and that this error-handling code had been largely removed.

It has?

Reply:
I still see some, but I guess at least a portion of it must have been otherwise what does @jasondegraw's comment mean?

Reply:
I stopped adding the `else` to my code, but I think there is a lot still there that's left over from pre-epJSON days. Some is hiding in the pre-epJSON style input.

Reply:
Okay. At some point in the near future, we will start to see the second and subsequent phases of the `enum` transition roll in, including use of `constexpr std::array<std::string_view>`, `getEnumerationValue`, and `switch\case` instead of string-based `if-else-if` ladders. We can remove these else clauses at that time. No point in doing the same checks and printing similar errors twice.

Reply:
I appreciate all this conversation.  For the sake of this PR, I was asked to add in GSL::span so that we could start building on it.  I put in this function as a demonstration only, I'm not even using this new function.  If I'm picking up the mood of this thread, it feels like this PR doesn't necessarily need to change, but when we start using this function in another branch, we may want to consider some slight modification.  

If I am correct there, can we get this conversation moved to a [Discussion](https://github.com/NREL/EnergyPlus/discussions/new) or something and point back to this comment block?  However, if I need to change it here to get this PR in, let me know and I'm happy to alter it.

Reply:
No need to change the PR. The last several messages have been about how to use the PR in the (near) future. Sorry.

Reply:
:+1: